lib/api/apiUtils/quotas/quotaUtils.js

```diff
@@ -225,17 +246,20 @@ function validateQuotas(request, bucket, account, apiNames, apiMethod, inflight,
             return done();
         });
     }, err => {
+        const responseTimeInNs = Number(process.hrtime.bigint() - requestStartTime);
         if (err) {
             log.warn('Error getting metrics from the quota service, allowing the request', {
                 error: err.name,
                 description: err.message,
             });
         }
         if (actionWithDataDeletion[apiMethod]) {
-            return callback();
+            monitorQuotaEvaluation(apiMethod, type, 200, responseTimeInNs / 1e9);
         } else if (bucketQuotaExceeded || accountQuotaExceeded) {
+            monitorQuotaEvaluation(apiMethod, type, 429, responseTimeInNs / 1e9);
             return callback(errors.QuotaExceeded);
         }
+        monitorQuotaEvaluation(apiMethod, type, 200, responseTimeInNs / 1e9);
```

Comment:
this computes the metrics for the call to `validateQuotas` ; however, it does not account for the eventual "cleanup" which may happen at the end of failed request...
→ to be "precise", we should store the value here (in the request), always call the cleanup callback, and there also add the cleanup time (if any) before observing the value...
→ otherwise, maybe we should do a second measure if the cleanup is executed.... but then maybe the metric is not correctly named? or maybe the cleanup can be observed via another metric?
→ or maybe this is correct, as we are really only interested in the actual _evaluation_ part, and not the "total cost of quota" ? That may be fine, but should be made clear in the comments (or description?) of the metric